edgelet/edgelet-docker/src/notary.rs

```diff
@@ -0,0 +1,90 @@
+use std::path::{Path,PathBuf};
+use log::{info, debug};
+use serde_json::json;
+use std::fs;
+use std::fs::File;
+use crate::{Error,ErrorKind};
+use failure::ResultExt;
+
+pub fn notary_init(homedir : &Path, hostname : String, path : PathBuf) -> Result<(), Error> {
+  // Validate inputs
+  if hostname.is_empty() {
+    return Err(ErrorKind::InitializeNotary("hostname is empty".to_owned()).into());
+  }
+
+  if !path.exists() {
+    return Err(ErrorKind::InitializeNotary(format!("root ca at path {} does not exist", path.display())).into());
+  }
+
+  // Create notary home directory
+  let notary_dir = homedir.join("notary");
+
+  // Create trust directory
+  let mut trust_dir = notary_dir;
+  // hash hostname  and then santize and concat
+  // concat - + () 
+  trust_dir.push(&hostname);
+  info!("Trust directory is {}", trust_dir.display());
+  if let Err(err) = fs::remove_dir_all(&trust_dir){
+    if err.kind() != std::io::ErrorKind::NotFound {
```

Comment:
So, on a clean machine, the notary direction will not be found. In that case, it is expected, right?

Reply:
Yes in a clean machine - we won't have the path and it is expected. Thats why the check is to make sure the error kind is not "Not found" category.